presto-native-execution/presto_cpp/main/TaskManager.cpp

```diff
@@ -51,11 +51,9 @@ static void maybeSetupTaskSpillDirectory(
 
   const auto includeNodeInSpillPath =
       SystemConfig::instance()->includeNodeInSpillPath();
-  auto nodeConfig = NodeConfig::instance();
   const auto taskSpillDirPath = TaskManager::buildTaskSpillDirectoryPath(
       baseSpillDirectory,
-      nodeConfig->nodeInternalAddress(),
-      nodeConfig->nodeId(),
+      NodeConfig::instance(),
```

Comment:
We can invoke `NodeConfig::instance()` directly inside the `buildTaskSpillDirectoryPath`
in the check `if (includeNodeInSpillPath)`

Reply:
Thanks Deepak! I pushed a new change.